Makefile

```diff
@@ -6,7 +6,15 @@
 # %LICENSE%
 #
 
-VERSION="3.3 rev: $(shell git rev-parse --short HEAD)"
+# DESC will be in format v<VERSION> <#COMMITS SINCE TAG> g<CURRENT COMMIT HASH>[ dirty]
+DESC := $(subst -, ,$(shell git describe --dirty --abbrev=8 --tags --long --match "v[[:digit:]].[[:digit:]]*.[[:digit:]]*"))
+
+VERSION := $(subst g,,$(word 3,$(DESC)))\ +$(word 2,$(DESC))
+ifneq ($(word 4,$(DESC)),)
+	VERSION := $(VERSION)\ $(word 4,$(DESC))
```

Comment:
please use 2 spaces for indentation in Makefiles (TAB is a default RULEPREFIX in Makefile and may result in strange errors).